src/ispyb/cli/job.py

```diff
@@ -0,0 +1,467 @@
+    for p in options.parameters:
+        if ":" not in p:
+            sys.exit("Invalid parameter specification: " + p)
+        parameters.append(p.split(":", 1))
+
+    if zocalo:
+        trigger_variables = []
+        for p in options.triggervariables:
+            if ":" not in p:
+                sys.exit("Invalid trigger variable specification: " + p)
+            trigger_variables.append(p.split(":", 1))
+
+    jp = i.mx_processing.get_job_params()
+    # _job_params = StrictOrderedDict([('id', None), ('datacollectionid', None), ('display_name', None), ('comments', None), ('recipe', None), ('automatic', None)])
+    jp["automatic"] = options.source == "automatic"
+    jp["comments"] = options.comment
+    jp["datacollectionid"] = dcid or sweeps[0][0]
+    jp["display_name"] = options.display
+    jp["recipe"] = options.recipe
+    print("Creating database entries...")
+
+    jobid = i.mx_processing.upsert_job(list(jp.values()))
+    print(f"  JobID={jobid}")
+    for key, value in parameters:
+        jpp = i.mx_processing.get_job_parameter_params()
+        # _job_parameter_params = StrictOrderedDict([('id', None), ('job_id', None), ('parameter_key', None), ('parameter_value', None)])
+        jpp["job_id"] = jobid
+        jpp["parameter_key"] = key
+        jpp["parameter_value"] = value
+        jppid = i.mx_processing.upsert_job_parameter(list(jpp.values()))
+        print(f"  JPP={jppid}")
+
+    for sweep in sweeps:
+        jisp = i.mx_processing.get_job_image_sweep_params()
+        # _job_image_sweep_params = StrictOrderedDict([('id', None), ('job_id', None), ('datacollectionid', None), ('start_image', None), ('end_image', None)])
+        jisp["job_id"] = jobid
+        jisp["datacollectionid"] = sweep[0]
+        jisp["start_image"] = sweep[1]
+        jisp["end_image"] = sweep[2]
+        jispid = i.mx_processing.upsert_job_image_sweep(list(jisp.values()))
+        print(f"  JISP={jispid}")
+
+    print(f"All done. Processing job {jobid} created")
+    print()
+    if zocalo:
+        if options.trigger:
+            go_call = ["zocalo.go", "-p", str(jobid)]
+            for kv in trigger_variables:
+                go_call.append("--set=%s=%s" % (kv[0], kv[1]))
+            result = procrunner.run(go_call)
+            if result["exitcode"] or result["stderr"]:
+                sys.exit("Error triggering processing job")
+            print("Successfully triggered processing job")
+            print()
+
+        else:
+            print("To trigger the processing job you now need to run:")
+            print(f"  zocalo.go -p {jobid}")
+            print()
+
+    return jobid
+
+
+def run():
+    parser = OptionParser(
+        usage="ispyb.job [options] rpid",
+        description="Command line tool to manipulate ISPyB processing table entries.",
+    )
+
+    available_recipes = filter(
+        lambda r: r.startswith("ispyb-") and r.endswith(".json"),
+        os.listdir("/dls_sw/apps/zocalo/live/recipes"),
```

Comment:
this is catered for by zocalo.configuration, but thats still to be reviewed

c.f.
https://github.com/DiamondLightSource/python-zocalo-ispyb/blob/main/zocalo_ispyb/cli/ispyb_job.py#L138

https://github.com/DiamondLightSource/python-zocalo/pull/52/files#diff-a00122aee2a4c3b731fcc1d940db7a3737c99d09611afd3d909a5a453af711ebR3

Reply:
Made this code block optional :+1: